analysis/filter/MergeFilter/src/main/kotlin/de/maibornwolff/codecharta/filter/mergefilter/LargeMerge.kt

```diff
@@ -0,0 +1,54 @@
+package de.maibornwolff.codecharta.filter.mergefilter
+
+import de.maibornwolff.codecharta.model.BlacklistItem
+import de.maibornwolff.codecharta.model.Edge
+import de.maibornwolff.codecharta.model.Node
+import de.maibornwolff.codecharta.model.NodeType
+import de.maibornwolff.codecharta.model.Project
+
+class LargeMerge {
+    companion object {
+        fun packageProjectInto(project: Project, prefix: String): Project {
+            val modifiedProject = Project(
+                project.projectName,
+                moveNodesIntoFolder(project.rootNode, prefix),
+                project.apiVersion,
+                addFolderToEdgePaths(project.edges, prefix),
+                project.attributeTypes,
+                project.attributeDescriptors,
+                addFolderToBlackListPaths(project.blacklist, prefix)
+            )
+            return modifiedProject
+        }
+
+        private fun moveNodesIntoFolder(root: Node, folderName: String): List<Node> {
+            val newRoot = Node(
+                name = folderName,
+                type = NodeType.Folder,
+                children = root.children
+            )
+            val mutableRoot = root.toMutableNode()
+            mutableRoot.children = mutableSetOf(newRoot.toMutableNode())
+            return listOf(mutableRoot.toNode())
+        }
+
+        private fun addFolderToEdgePaths(edges: List<Edge>, folderName: String): List<Edge> {
+            edges.forEach {
+                it.fromNodeName = insertFolderIntoPath(it.fromNodeName, folderName)
+                it.toNodeName = insertFolderIntoPath(it.toNodeName, folderName)
+            }
+            return edges
+        }
+
+        private fun addFolderToBlackListPaths(blacklist: List<BlacklistItem>, folderName: String): List<BlacklistItem> {
+            blacklist.forEach {
+                it.path = insertFolderIntoPath(it.path, folderName)
+            }
+            return blacklist
+        }
```

Comment:
These functions modify there parameters. Is that on purpose? if not, better to create a new array with map and copy while modifying.
something like this 
```
return edges.map { edge ->
        edge.copy(
            fromNodeName = insertFolderIntoPath(edge.fromNodeName, folderName),
            toNodeName = insertFolderIntoPath(edge.toNodeName, folderName)
        )
    }
```

Reply:
Oh yeah, but it's better for the future if we keep the methods pure. Otherwise, it could lead to weird behavior during a refactor. Try it out if it works change it. If not that's alright.